lib/pbench/server/database/alembic/env.py

```diff
@@ -0,0 +1,112 @@
+"""
+This file is auto generated when we run `alembic init alembic` but modified according to our needs.
+This Python script runs whenever the alembic migration tool is invoked.
+It contains instructions to configure and generate a SQLAlchemy engine,
+procure a connection from that engine along with a transaction, and then
+invoke the migration engine, using the connection as a source of database connectivity.
+"""
+import sys
+import logging
+from logging.config import fileConfig
+
+from alembic import context
+
+from pbench.common.exceptions import BadConfig, ConfigFileNotSpecified
+from pbench.server.database.database import Database
+from pbench.common.logger import get_pbench_logger
+from pbench.server.api import get_server_config
+
+# this is the Alembic Config object, which provides
+# access to the values within the .ini file in use.
+config = context.config
+
+# Interpret the config file for Python logging.
+# This line sets up loggers basically.
+fileConfig(config.config_file_name)
+
+# Add syslog handler to send logs to journald
+log = logging.getLogger("alembic")
+handler = logging.handlers.SysLogHandler("/dev/log")
+log.addHandler(handler)
+
+# add your model's MetaData object here
+# for 'autogenerate' support
+# from myapp import mymodel
+# target_metadata = mymodel.Base.metadata
+target_metadata = [Database.Base.metadata]
+
+# other values from the config, defined by the needs of env.py,
+# can be acquired:
+# my_important_option = config.get_main_option("my_important_option")
+# ... etc.
+
+
+def run_migrations_offline():
+    """Run migrations in 'offline' mode.
+
+    This configures the context with just a URL
+    and not an Engine, though an Engine is acceptable
+    here as well.  By skipping the Engine creation
+    we don't even need a DBAPI to be available.
+
+    Calls to context.execute() here emit the given string to the
+    script output.
+
+    """
+    # url = config.get_main_option("sqlalchemy.url")
+    try:
+        server_config = get_server_config()
+        logger = get_pbench_logger(__name__, server_config)
+    except (ConfigFileNotSpecified, BadConfig) as e:
+        print(e)
+        sys.exit(1)
```

Comment:
What does this code achieve (ditto lines 89-91)?  In particular, how is the handling of these two exceptions different from the handling of any other exceptions raised in this `try` block?

That is, it seems like we should either be catching all exceptions here or catching none -- I don't see how handling just these two in this way improves over the all or none approaches.

Reply:
fixed